packages/protocol/contracts/governance/Proposals.sol

```diff
@@ -0,0 +1,356 @@
+   * @return The specified transaction.
+   */
+  function getTransaction(
+    Proposal storage proposal,
+    uint256 index
+  )
+    public
+    view
+    returns (uint256, address, bytes memory)
+  {
+    require(index < proposal.transactions.length);
+    Transaction storage transaction = proposal.transactions[index];
+    return (transaction.value, transaction.destination, transaction.data);
+  }
+
+  /**
+   * @notice Returns an unpacked proposal struct with its transaction count.
+   * @param proposal The proposal struct.
+   * @return The unpacked proposal with its transaction count.
+   */
+  function unpack(
+    Proposal storage proposal
+  )
+    internal
+    view
+    returns (address, uint256, uint256, uint256)
+  {
+    return (
+      proposal.proposer,
+      proposal.deposit,
+      proposal.timestamp,
+      proposal.transactions.length
+    );
+  }
+
+  /**
+   * @notice Returns the referendum vote totals for a proposal.
+   * @param proposal The proposal struct.
+   * @return The yes, no, and abstain vote totals.
+   */
+  function getVoteTotals(
+    Proposal storage proposal
+  )
+    internal
+    view
+    returns (uint256, uint256, uint256)
+  {
+    return (proposal.votes.yes, proposal.votes.no, proposal.votes.abstain);
+  }
+
+  /**
+   * @notice Returns whether or not a proposal has been approved.
+   * @param proposal The proposal struct.
+   * @return Whether or not the proposal has been approved.
+   */
+  function isApproved(Proposal storage proposal) internal view returns (bool) {
+    return proposal.approved;
+  }
+
+  /**
+   * @notice Returns whether or not a proposal exists.
+   * @param proposal The proposal struct.
+   * @return Whether or not the proposal exists.
+   */
+  function exists(Proposal storage proposal) internal view returns (bool) {
+    return proposal.timestamp > 0;
+  }
+
+  // call has been separated into its own function in order to take advantage
+  // of the Solidity's code generator to produce a loop that copies tx.data into memory.
+  /**
+   * @notice Executes a function call.
+   * @param value The value of Celo Gold to be sent with the function call.
+   * @param destination The destination address of the function call.
+   * @param dataLength The length of the data to be included in the function call.
+   * @param data The data to be included in the function call.
+   */
+  function externalCall(
+    address destination,
+    uint value,
+    uint dataLength,
+    bytes memory data
+  )
+    private
+    returns (bool)
+  {
+    bool result;
+    /* solhint-disable no-inline-assembly */
+    assembly {
+      /* solhint-disable max-line-length */
+      let x := mload(0x40)   // "Allocate" memory for output (0x40 is where "free memory" pointer is stored by convention)
+      let d := add(data, 32) // First 32 bytes are the padded length of data, so exclude that
+      result := call(
+        sub(gas, 34710),   // 34710 is the value that solidity is currently emitting
```

Comment:
Even with the comment it's not entirely clear to me why we need this. Could you explain?

Reply:
This is moved over from the original implementation in Governance. My guess is that trials were run to examine the added gas cost of executing a transaction through Governance and that was the number arrived at. Perhaps getting a clarification from @asaj and putting it in the code for posterity would be best

Reply:
I guess one question is if this is even still useful given that we can execute multiple proposals - saving that extra gas in the first transaction doesn't necessarily mean we'll be able to successfully execute the whole proposal, since there might not be enough gas for the next transaction. 

@asaj, thoughts?